lib/asciidoctor-revealjs/highlightjs.rb

```diff
@@ -33,7 +35,21 @@ def docinfo location, doc, opts
           else
             theme_href = "#{revealjsdir}/lib/css/zenburn.css"
           end
-          %(<link rel="stylesheet" href="#{theme_href}"#{opts[:self_closing_tag_slash]}>)
+
+          unless (asset_uri_scheme = (doc.attr 'asset-uri-scheme', 'https')).empty?
+            asset_uri_scheme = %(#{asset_uri_scheme}:)
+          end
+          # use cdn.jsdelivr.net instead of cdnjs.com because it contains the latest version of highlight.js with additional languages
+          # please note that the bundled version of highlight.js in reveal.js contains 175 languages
+          # the prebuilt version of highlight.js hosted on cdn.jsdelivr.net contains only 34 commonly used languages.
+          # if you want additional languages, use "highlightjs-languages".
+          # QUESTION: should we also provide a bundled version of highlight.js with all the languages included?
+          cdn_base = %(#{asset_uri_scheme}//cdn.jsdelivr.net)
+          plugin_highlight_path = File.join(File.dirname(__FILE__), "../../assets/plugin-highlight.js")
+          base_url = doc.attr 'highlightjsdir', %(#{cdn_base}/gh/highlightjs/cdn-release@#{HIGHLIGHT_JS_VERSION}/build)
+          %(<link rel="stylesheet" href="#{theme_href}"#{opts[:self_closing_tag_slash]}>
+<script src="#{base_url}/highlight.min.js"></script>
+#{(doc.attr? 'highlightjs-languages') ? ((doc.attr 'highlightjs-languages').split ',').map {|lang| %[<script src="#{base_url}/languages/#{lang.lstrip}.min.js"></script>\n] }.join : ''}<script>#{File.read(plugin_highlight_path)}</script>)
```

Comment:
@obilodeau I think we should avoid using `File.read` as it will cause issues in Node. Maybe we should just embed the JavaScript code in the Ruby file (it's ~100 lines)?

Reply:
Either way, I don't mind.

It's funny that you review your own PR :laughing:

Reply:
I tend to forget things so it's a kind of memo 😄